libs/blocks/youtube/youtube.js

```diff
@@ -1,36 +1,107 @@
-import { createIntersectionObserver, isInTextNode } from '../../utils/utils.js';
+// part of the code is an optimized version of lite-youtube-embed -> https://github.com/paulirish/lite-youtube-embed
+import { createIntersectionObserver, createTag, isInTextNode, loadLink } from '../../utils/utils.js';
+
+class LiteYTEmbed extends HTMLElement {
+  connectedCallback() {
+    this.videoId = this.getAttribute('videoid');
+    const playBtnEl = createTag('button', { type: 'button', class: 'lty-playbtn' });
+    this.append(playBtnEl);
+    this.playLabel = this.getAttribute('playlabel') || 'Play';
+    this.style.backgroundImage = `url("https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg")`;
+    this.style.backgroundSize = 'cover';
+    this.style.backgroundPosition = 'center';
+    const playBtnLabelEl = createTag('span', { class: 'lyt-visually-hidden' });
+    playBtnLabelEl.textContent = this.playLabel;
+    playBtnEl.append(playBtnLabelEl);
+    this.addEventListener('pointerover', LiteYTEmbed.warmConnections, { once: true });
+    this.addEventListener('click', this.addIframe);
+    this.needsYTApiForAutoplay = navigator.vendor.includes('Apple') || navigator.userAgent.includes('Mobi');
+  }
+
+  static warmConnections() {
+    if (LiteYTEmbed.preconnected) return;
+    LiteYTEmbed.preconnected = true;
+    ['www.youtube-nocookie.com',
+      'www.google.com',
+      'googleads.g.doubleclick.net',
+      'static.doubleclick.net',
+    ].forEach((url) => loadLink(`https://${url}`, { rel: 'preconnect' }));
+  }
+
+  static loadYouTubeAPI() {
+    return new Promise((resolve) => {
+      if (window.YT && window.YT.Player) {
+        resolve();
+        return;
+      }
+      const tag = createTag('script', { src: 'https://www.youtube.com/iframe_api' });
+      window.onYouTubeIframeAPIReady = resolve;
```

Comment:
I will assume this is used by the iframe_api script

Reply:
Yes, it is